File: Programming Basics with JS/08. More Exersices/movingOut.js
function solve (input) {
    i = 0;
    let width = Number(input[i]);
    i++;
    let length = Number(input[i]);
    i++;
    let highth = Number(input[i]);
    i++;

    let isFreeSpace = true;
    let cubicMetres = width * length * highth;
    let command = input[i];
    i++;

    while(command !== "Done"){
        let box = Number(command);
        cubicMetres -= box;

        if(cubicMetres<0){
            isFreeSpace = false;
            console.log(`No more free space! You need ${Math.abs(cubicMetres)} Cubic meters more.`);
            break;
        }

        command = input[i];
        i++;
    }

    if(isFreeSpace){
        console.log(`${cubicMetres} Cubic meters left.`);
    }

}

solve (["10",
"1",
"2",
"4",
"6",
"Done"])

File: Programming Basics with Java/05. While Loop - Lab/src/MovingOut.java
import java.util.Scanner;

public class MovingOut {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int width = Integer.parseInt(scan.nextLine());
        int length = Integer.parseInt(scan.nextLine());
        int height = Integer.parseInt(scan.nextLine());
        int freeSpace = width * length * height;
        boolean isFull = false;
        String input = scan.nextLine();

        while (input.equals("Done") == false){
            int volume = Integer.parseInt(input);
            freeSpace -= volume;

            if(freeSpace <= 0){
                isFull = true;
                break;
            }

            input = scan.nextLine();
        }

        if(isFull){
            System.out.printf("No more free space! You need %d Cubic meters more.", Math.abs(freeSpace));
        } else {
            System.out.printf("%d Cubic meters left.", freeSpace);
        }

    }
}